src/Feature/StringifyProviderInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * Copyright BibLibre, 2017
+ *
+ * This software is governed by the CeCILL license under French law and abiding
+ * by the rules of distribution of free software.  You can use, modify and/ or
+ * redistribute the software under the terms of the CeCILL license as circulated
+ * by CEA, CNRS and INRIA at the following URL "http://www.cecill.info".
+ *
+ * As a counterpart to the access to the source code and rights to copy, modify
+ * and redistribute granted by the license, users are provided only with a
+ * limited warranty and the software's author, the holder of the economic
+ * rights, and the successive licensors have only limited liability.
+ *
+ * In this respect, the user's attention is drawn to the risks associated with
+ * loading, using, modifying and/or developing or reproducing the software by
+ * the user in light of its specific status of free software, that may mean that
+ * it is complicated to manipulate, and that also therefore means that it is
+ * reserved for developers and experienced professionals having in-depth
+ * computer knowledge. Users are therefore encouraged to load and test the
+ * software's suitability as regards their requirements in conditions enabling
+ * the security of their systems and/or data to be ensured and, more generally,
+ * to use and operate it in the same conditions as regards security.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL license and that you accept its terms.
+ */
+
+namespace Search\Feature;
+
+interface StringifyProviderInterface
+{
+    public function stringifyData();
```

Comment:
Parameters are missing and the interface is not used anywhere.

All classes that implement `stringifyData` should implement this interface and code that calls `stringifyData` should check if the corresponding object's class implement this interface. Otherwise it is useless.

A comment about the parameters and the expected return value would be nice.

A suggestion about naming (feel free to disagree):
* "stringify" feels wrong as the method now returns an array (of strings, yes, so it is still kind of true)
* "data" is very generic and I think we can find a better (more precise) term
* the purpose of this method is to create a human readable description/summary of the search query, so something like `describeQuery`, `summarizeQuery`, `getQueryDescription`, or `getQuerySummary` would be more precise.
* if you agree, don't forget to update the interface name